src/transformers/models/auto/tokenization_auto.py

```diff
@@ -590,6 +590,14 @@ def tokenizer_class_from_name(class_name: str):
     return None
 
 
+def has_pretrainedfast(class_name: str):
+    for module_name, tokenizers in TOKENIZER_MAPPING_NAMES.items():
+        if class_name in tokenizers and "PreTrainedTokenizerFast" in tokenizers:
+            return PreTrainedTokenizerFast
+
+    return None
```

Comment:
I don't think we need a one liner, nor do we need a for loop! We can just get tokenizer_mapping_name["class_name"][1]

Reply:
yeah the problem is we don't actually have the class name at this point, would have to regex to get 'siglip'

Reply:
updated pr